mobilecoind/src/bin/main.rs

```diff
@@ -27,8 +27,17 @@ fn main() {
     let _sentry_guard = mc_common::sentry::init();
     let (logger, _global_logger_guard) = create_app_logger(o!());
 
-    let _tracer = mc_util_telemetry::setup_default_tracer(env!("CARGO_PKG_NAME"))
-        .expect("Failed setting telemetry tracer");
+    // Telemetry is disabled if MC_TELEMETRY is set to "0"
+    let telemetry_enabled = !std::env::var("MC_TELEMETRY")
+        .map(|val| val == "0")
+        .unwrap_or(false);
+
+    let _tracer = if !telemetry_enabled {
+        let _tracer = Some(mc_util_telemetry::setup_default_tracer(env!("CARGO_PKG_NAME"))
+            .expect("Failed setting telemetry tracer"));
+    } else {
+        None
+    }
```

Comment:
This should fix the build
```suggestion
        Some(mc_util_telemetry::setup_default_tracer(env!("CARGO_PKG_NAME"))
            .expect("Failed setting telemetry tracer"));
    } else {
        None
    };
```